File: semestr-5/Algorytmy i struktury danych 1/selection_sort.py

def selection_sort(tab):
    if len(tab) == 0:
        return tab
    for i in range(len(tab)):
        min_index = i
        for j in range(i, len(tab)):
            if tab[j] < tab[min_index]:
                min_index = j
        tab[i], tab[min_index] = tab[min_index], tab[i]
        
        print(min_index, tab)
    return tab


tab = [11,5,7,2,15,4,6,12,87,36,22,17,91,92,96,97]

selection_sort(tab)

File: semestr-2/Podstawy programowania/ppr/ANTy_PP/_Pomoce/selectionsort.c
/* Program sortuje tablice nierosnaco algorytmem SelectionSort.
(sortowanie przez selekcje) */
#include <stdio.h>
#include <stdlib.h>
#define rozm 20
int max_od(int tab[], int n, int k)
{
 int i, max=tab[k], poz=k;
for(i=k+1;i<n;i++)
if(max<tab[i]){ //wystarczy zmienic znak '<' na '>'
max=tab[i]; //zeby otrzymac ciag niemalejacy
poz=i;
}
return poz;
}
void zamien(int tab[], int i, int j)
{
 int temp;
temp=tab[i]; tab[i]=tab[j]; tab[j]=temp;
}
void SelectionSort(int tab[], int n)
{
 int i,poz;
for(i=0;i<n-1;i++)
if(tab[i]!=tab[poz=max_od(tab,n,i)])
zamien(tab,i,poz);
}
void wyp_los(int tab[], int n, int k)
{
 int i; for(i=0;i<n;i++) tab[i]=random(k);
}
void wypisz(int tab[], int n)
{
 int i; for(i=0;i<n;i++) printf("%i, ",tab[i]);
printf("\n");
}
void main()
{
 int t[rozm];
randomize();
wyp_los(t,rozm,rozm);
wypisz(t,rozm);
SelectionSort(t,rozm);
wypisz(t,rozm);
}
